src/sos/objectstorage.go

```diff
@@ -0,0 +1,120 @@
+	k8sClients     *[]src.K8sClient
+	exoscaleClient *egoscale.Client
+	bucketDetails  []BucketDetail
+}
+
+type BucketDetail struct {
+	ClusterHostname string
+	Namespace       string
+	Organization    string
+	BucketName      string
+}
+
+func NewObjectStorage(exoscaleClient *egoscale.Client, k8sClients *[]src.K8sClient) ObjectStorage {
+	return ObjectStorage{
+		exoscaleClient: exoscaleClient,
+		k8sClients:     k8sClients,
+	}
+}
+
+func (o *ObjectStorage) Execute(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.V(1).Info("Running metrics collector by step", "config", o)
+	ctrl.SetLogger(log)
+
+	p := pipeline.NewPipeline[context.Context]()
+	p.WithSteps(
+		p.NewStep("Fetch managed buckets", o.fetchManagedBuckets),
+		p.NewStep("Get bucket usage", o.getBucketUsage),
+	)
+	return p.RunWithContext(ctx)
+}
+
+func (o *ObjectStorage) getBucketUsage(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	log.V(1).Info("Fetching bucket usage from Exoscale")
+	resp, err := o.exoscaleClient.ListSosBucketsUsageWithResponse(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, bucketUsage := range *resp.JSON200.SosBucketsUsage {
+		name := *bucketUsage.Name
+		zoneName := *bucketUsage.ZoneName
+		size := *bucketUsage.Size
+		createdAt := *bucketUsage.CreatedAt
+		bucketIndex := slices.IndexFunc(o.bucketDetails, func(ni BucketDetail) bool { return ni.BucketName == name })
+		if bucketIndex == -1 {
+			log.V(1).Info("Cannot find bucket from clusters", name)
+			continue
+		}
+		fmt.Printf("name: %s, zoneName: %s, size: %d, createdAt: %s, namespace info: %s\n", name, zoneName, size, createdAt, o.bucketDetails[bucketIndex])
+	}
+	return nil
+}
+
+func (o *ObjectStorage) fetchManagedBuckets(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+	for _, k8sClient := range *o.k8sClients {
+		buckets := exoscalev1.BucketList{}
+		err := k8sClient.List(ctx, &buckets)
+		if err != nil {
+			log.V(1).Error(err, "Cannot list buckets")
+			return err
+		}
+		bucketDetailsByCluster, err := matchOrgAndNamespaceByBucket(ctx, &buckets, k8sClient.ClusterHostname)
+		if err != nil {
+			log.V(1).Error(err, "Cannot fetch bucket details from buckets")
+			return err
+		}
+
+		// add BucketDetail from each cluster to one single slice
+		o.bucketDetails = append(o.bucketDetails, *bucketDetailsByCluster...)
+	}
+	return nil
+}
+
+func matchOrgAndNamespaceByBucket(ctx context.Context, buckets *exoscalev1.BucketList, clusterHostname string) (*[]BucketDetail, error) {
+	log := ctrl.LoggerFrom(ctx)
+	var bucketDetails []BucketDetail
+	for _, bucket := range buckets.Items {
+		bucketDetail := BucketDetail{
+			BucketName:      bucket.Spec.ForProvider.BucketName,
+			ClusterHostname: clusterHostname,
+		}
+		if organization, exist := bucket.ObjectMeta.Labels[organizationLabel]; exist {
+			bucketDetail.Organization = organization
+		} else {
+			// cannot get organization from bucket
+			log.V(1).Info("Organization label is missing in bucket", bucket.Name)
```

Comment:
This is a reasonably big enough reason to not have it as a debug log IMO
```suggestion
			log.Info("Organization label is missing in bucket", bucket.Name)
```